tests/utils.py

```diff
@@ -94,7 +96,39 @@ def get_llama_cpp_model(model_name, caching=False, **kwargs):
     key = model_name + "_" + str(caching) + "_" + str(kwargs)
     if key not in llama_cpp_model_cache:
         llama_cpp_model_cache[key] = guidance.models.LlamaCpp(
-            model_name, caching=caching, **kwargs
+            model_name, **kwargs
         )
 
     return llama_cpp_model_cache[key]
+
+
+azure_guidance_model_cache = {}
+azure_guidance_defaults = {}
+
+
+def get_azure_guidance_model(model_name, caching=False, **kwargs):
+    """Get Azure Guidance LLM with model reuse."""
+
+    if (
+        model_name is None
+        or isinstance(model_name, str)
+        and len(model_name.strip()) == 0
+    ):
+        model_name = os.getenv("AZURE_GUIDANCE_URL", "")
+        if len(model_name.strip()) == 0:
+            pytest.skip("No Azure Guidance model found.")
+
+    kwargs = kwargs.copy()
+    for key, val in azure_guidance_defaults.items():
+        if key not in kwargs:
+            kwargs[key] = val
+
+    # we cache the models so lots of tests using the same model don't have to
+    # load it over and over again
```

Comment:
Do we really need this? I think that these models are _really_ lightweight.

Reply:
Probably not, but I'm following the code format of the other models in the same file above, which are more expensive.